go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -165,59 +170,149 @@ func (oc *Controller) updateEgressFirewallWithRetry(egressfirewall *egressfirewa
 	})
 }
 
-func (ef *egressFirewall) addACLToJoinSwitch(joinSwitches []string, hashedAddressSetNameIPv4, hashedAddressSetNameIPv6 string) error {
+func (oc *Controller) applyEgressFirewall(joinSwitches []string, hashedAddressSetNameIPv4, hashedAddressSetNameIPv6 string, ef *egressFirewall) error {
 	for _, rule := range ef.egressRules {
-		var match string
 		var action string
+		var matchTargets []matchTarget
 		if rule.access == egressfirewallapi.EgressFirewallRuleAllow {
 			action = "allow"
 		} else {
 			action = "drop"
 		}
+		if rule.to.cidrSelector != "" {
+			if utilnet.IsIPv6CIDRString(rule.to.cidrSelector) {
+				matchTargets = []matchTarget{{matchKindV6CIDR, rule.to.cidrSelector}}
+			} else {
+				matchTargets = []matchTarget{{matchKindV4CIDR, rule.to.cidrSelector}}
 
-		ipAddress, _, err := net.ParseCIDR(rule.to.cidrSelector)
-		if err != nil {
-			return fmt.Errorf("error rule.to.cidrSelector %s is not a valid CIDR (%+v)", rule.to.cidrSelector, err)
-		}
-		if !utilnet.IsIPv6(ipAddress) {
-			match = fmt.Sprintf("match=\"ip4.dst == %s && ip4.src == $%s\"", rule.to.cidrSelector, hashedAddressSetNameIPv4)
+			}
 		} else {
-			match = fmt.Sprintf("match=\"ip6.dst == %s && ip6.src == $%s\"", rule.to.cidrSelector, hashedAddressSetNameIPv6)
+			// rule based on DNS NAME
+			if rule.access == egressfirewallapi.EgressFirewallRuleDeny {
+				return fmt.Errorf("do not use dnsName as part of a deny rule this is insecure")
+			}
+			dnsNameAddressSets, err := oc.egressFirewallDNS.Add(ef.namespace, rule.to.dnsName)
+			if err != nil {
+				return fmt.Errorf("error with EgressFirewallDNS - %v", err)
+			}
+			if dnsNameAddressSets.GetIPv4HashName() != "" {
+				matchTargets = append(matchTargets, matchTarget{matchKindV4AddressSet, dnsNameAddressSets.GetIPv4HashName()})
+			}
+			if dnsNameAddressSets.GetIPv6HashName() != "" {
+				matchTargets = append(matchTargets, matchTarget{matchKindV6AddressSet, dnsNameAddressSets.GetIPv6HashName()})
+			}
 		}
-
-		if len(rule.ports) > 0 {
-			match = fmt.Sprintf("%s && ( %s )\"", match[:len(match)-1], egressGetL4Match(rule.ports))
+		match := generateMatch(hashedAddressSetNameIPv4, hashedAddressSetNameIPv6, matchTargets, rule.ports)
+		for _, joinSwitch := range joinSwitches {
+			err := createACLRule(defaultStartPriority-rule.id, match, action, ef.namespace, joinSwitch)
+			if err != nil {
+				return err
+			}
 		}
-		uuid, stderr, err := util.RunOVNNbctl("--data=bare", "--no-heading",
-			"--columns=_uuid", "find", "ACL", match, "action="+action,
-			fmt.Sprintf("external-ids:egressFirewall=%s", ef.namespace))
+	}
+	return nil
+}
+
+func createACLRule(priority int, match, action, namespace, joinSwitch string) error {
```

Comment:
This seems like it will duplicate the acls per join-switch, where previously there was only one acl applied to all switches? Or am I mis-reading the previous code?

Reply:
there needs to a an acl on all join switches. The code was written like that and I think I may have made it clearer that that is what is going on. Every node has a join switch and since we don't know where a pod is/would be the joinSwitchs need the acl for the namespace . It is a possible area of improvement to only put the ACLs on joinSwitches that currently have pods that the rule would apply too but it would it was unclear if that was worth the extra code complexity at this point.

Reply:
is it possible to put the same ACL on multiple switches?

Reply:
I think I read your comment wrong and now I understand and unless I am reading the code wrong that is how it is implemented maybe `createACLRule()` is a bad name. If you look at the implementation first it tries to find an ACL that matches the provided description and if it is not found create it otherwise add the ACL's uuid to the passed joinSwitch. Is that not what you are expecting?